pisa/stages/osc/pi_2nu.py

```diff
@@ -0,0 +1,147 @@
+        output_names = ()
+
+        # what are the keys used from the inputs during apply
+        input_apply_keys = ('weights',
+                            'nu_flux',
+                            'true_energy',
+                            'true_coszen',
+                           )
+
+        # what keys are added or altered for the outputs during apply
+        output_apply_keys = ('weights',
+                      )
+
+        # init base class
+        super(pi_2nu, self).__init__(
+            data=data,
+            params=params,
+            expected_params=expected_params,
+            input_names=input_names,
+            output_names=output_names,
+            debug_mode=debug_mode,
+            input_specs=input_specs,
+            calc_specs=calc_specs,
+            output_specs=output_specs,
+            input_apply_keys=input_apply_keys,
+            output_apply_keys=output_apply_keys,
+        )
+
+        assert self.input_mode is not None
+        assert self.calc_mode is not None
+        assert self.output_mode is not None
+
+    @profile
+    def apply_function(self):
+
+        theta23 = self.params.theta23.value.m_as('rad')
+        deltam31 = self.params.deltam31.value.m_as('eV**2')
+
+        for container in self.data:
+            if 'numu' in container.name:
+              apply_probs_vectorized(container['nu_flux'].get(WHERE),
+                                    theta23,
+                                    deltam31,
+                                    container['true_energy'].get(WHERE),
+                                    container['true_coszen'].get(WHERE),
+                                    1,
+                                    out=container['weights'].get(WHERE),
+                                   )
+
+            if 'nutau' in container.name:
+              apply_probs_vectorized(container['nu_flux'].get(WHERE),
+                                    theta23,
+                                    deltam31,
+                                    container['true_energy'].get(WHERE),
+                                    container['true_coszen'].get(WHERE),
+                                    3,
+                                    out=container['weights'].get(WHERE),
+                                   )
+            if 'nue' in container.name:
+              apply_probs_vectorized(container['nu_flux'].get(WHERE),
+                                    theta23,
+                                    deltam31,
+                                    container['true_energy'].get(WHERE),
+                                    container['true_coszen'].get(WHERE),
+                                    0,
+                                    out=container['weights'].get(WHERE),
+                                   )            
+            container['weights'].mark_changed(WHERE) 
+
+@myjit
+def calc_probs(t23, dm31, true_energy, true_coszen): #probability of numu to nutau
+
+    L1 = 19.
+    R = 6378.2 + L1
```

Comment:
Since here are two hard coded constants, could you add a description of what they are? I assume 19 is the assumed atmospheric production height in km?